src/V3Width.cpp

```diff
@@ -4265,6 +4265,13 @@ class WidthVisitor final : public VNVisitor {
         if (!nodep->dtypep() && m_vup->dtypeNullp()) {  // Get it from parent assignment/pin/etc
             nodep->dtypep(m_vup->dtypep());
         }
+        if (VN_IS(nodep->backp(), Arg) && VN_IS(nodep->backp()->backp(), MethodCall)) {
```

Comment:
AstArg should be passing down a dtype, so the previous `if` should see a m_vup->dtypeNullp() and set appropriately.

If it's the wrong data type, perhaps the code that iterates within the generated AstMethodCall is passing the wrong type?

Reply:
@wsnyder 
 In the visit(AstMethodCall* ) While iterating through the ARG expression, it is making m_vup  [WidthVP{SELF, BOTH}.p()](https://github.com/verilator/verilator/blob/87bd8fefa0e94e3563123d6a7832c30f3dac1b75/src/V3Width.cpp#L3132)

I think it should be like 
 WidthVP{nodep->fromp()->dtypep()->subDTypep(), BOTH}.p()

Reply:
Sounds right.